pxtblocks/blocklyloader.ts

```diff
@@ -585,7 +585,19 @@ namespace pxt.blocks {
     }
 
     function initField(i: any, ni: number, fn: pxtc.SymbolInfo, ns: pxtc.SymbolInfo, pre: string, right?: boolean, type?: string, nsinfo?: pxtc.SymbolInfo): any {
-        if (pre)
+        if (pre && pre.indexOf('`') > -1) {
+            // parse and create icon fields for every inline icon
+            let regex = /([^`]+|(`([^`]*)`))/gi;
+            let match: RegExpExecArray;
+            while (match = regex.exec(pre)) {
```

Comment:
tip: replace is easier to go through matches of a regex

Reply:
I tired that but for some reason I couldn't use it. I'll look into it again